lab_2_keywords_cooccurrence/main.py

```diff
@@ -123,8 +225,34 @@ def extract_candidate_keyword_phrases_with_adjoining(candidate_keyword_phrases:
 
     In case of corrupt input arguments, None is returned
     """
-    pass
-
+    if not (candidate_keyword_phrases and phrases):
+        return None
+    #тут будет проверка на тип
+    possible_pairs = {}
+    for i in range(len(candidate_keyword_phrases) - 1):
+        pair = candidate_keyword_phrases[i], candidate_keyword_phrases[i + 1]
+        if pair not in possible_pairs:
+            possible_pairs[pair] = 0
+        else:
+            possible_pairs[pair] += 1
+    appropriate_pairs = []
+    for pair in possible_pairs:
+        x = []
+        if possible_pairs[pair] > 1:
+            for i in pair:
+                i = ' '.join(i)
+                x.append(i)
+            if x:
+                appropriate_pairs.append(x)
+    phrases_with_stopwords = []
+    for phrase in phrases:
+        for elem in appropriate_pairs:
+            if (elem[0] and elem[1]) in phrase:
+                a = phrase.index(elem[0])
+                b = phrase.index(elem[1]) + len(elem[1])
+                needed_phrase = phrase[a:b]
+                phrases_with_stopwords.append(needed_phrase)
```

Comment:
what if there are phrases that consist of the same beginning and ending but have different stopwords in the middle? Or multiple stopwords?

Reply:
done(?)